server/src/main/java/org/opensearch/gateway/remote/model/RemoteClusterMetadataManifest.java

```diff
@@ -131,16 +132,18 @@ public ClusterMetadataManifest deserialize(final InputStream inputStream) throws
         return blobStoreFormat.deserialize(blobName, getNamedXContentRegistry(), Streams.readFully(inputStream));
     }
 
-    private int getManifestCodecVersion() {
+    // package private for testing
+    int getManifestCodecVersion() {
         assert blobName != null;
-        String[] splitName = blobName.split(DELIMITER);
+        String[] pathTokens = getBlobPathTokens();
```

Comment:
You can instead use `getBlobFileName()` method to get the file name

Reply:
Done